internal/test/builder/builders.go

```diff
@@ -922,6 +998,124 @@ func (b *TestBootstrapTemplateBuilder) Build() *unstructured.Unstructured {
 	return b.obj
 }
 
+// BootstrapMachinePoolTemplateBuilder holds the variables needed to build a generic BootstrapMachinePoolTemplate.
+type BootstrapMachinePoolTemplateBuilder struct {
```

Comment:
Instead of reuing the above `BootstrapTemplateBuilder` I went ahead an created this Builder to align it with [IInfrastructureMachinePoolTemplate](https://github.com/kubernetes-sigs/cluster-api/blob/main/internal/controllers/topology/cluster/scope/blueprint.go#L91). Which brings me back to my [original question](https://github.com/kubernetes-sigs/cluster-api/pull/9346#issuecomment-1702227355) why for infra there is `MachinePool` in the name, while for bootstrap it is not.

Reply:
Some concrete examples

core CAPI: MachineDeployment, Machinepool

bootstrap provider: 
generic names: BootstrapConfig, BootstrapConfigTemplate
kubeadm provider: KubeadmConfig, KubeadmConfigTemplate

infra provider,
generic names: InfraMachine, InfraMachineTemplate, InfraMachinePool
docker provider: DockerMachine, DockerMachineTemplate, DockerMachinePool

There is no BootstrapMachinePool, Machinepool just uses the regular BootstrapConfig (and ClusterClass references a BootstrapConfigTemplate)

Reply:
Ok, my bad. So that's the reason it explicitly has the MachinePool inside the field name. Will remove the BootstrapMachinePoolTemplateBuilder stuff again. Sorry, learning it the hard way :sweat_smile:

Reply:
No worries